src/gql/heart-monitor/heart-monitor.ts

```diff
@@ -237,25 +217,20 @@ export interface IHeartMonitor {
     fields?: DeepPartial<GQLStatsFields>
   ) => Promise<GqlOutStats>
 
-  readonly unbondings: (
-    args: GQLQueryGqlUnbondingsArgs,
-    fields?: DeepPartial<GQLUnbonding>
-  ) => Promise<GqlOutUnbondings>
-
   readonly users: (
     args: GQLQueryGqlUsersArgs,
     fields?: DeepPartial<GQLUser>
   ) => Promise<GqlOutUsers>
 
-  readonly validators: (
-    args: GQLQueryGqlValidatorsArgs,
-    fields?: DeepPartial<GQLValidator>
-  ) => Promise<GqlOutValidators>
-
   readonly wasm: (
     args: QueryWasmArgs,
     fields?: DeepPartial<GqlWasmFields>
   ) => Promise<GqlOutWasm>
+
+  readonly staking: (
+    args: QueryStakingArgs,
+    fields?: DeepPartial<GQLStakingFields>
```

Comment:
This should just be `DeepPartial<GQLStaking>`

Reply:
GQLStaking fields are ReadonlyArray of some objects (e.g. ReadonlyArray<GQLValidator>) 
This is not the same as defined in GQLStakingFields. At least convertObjectToPropertiesString doesn't know how to convert array of objects to the string representation of selected fields.

Reply:
convertObjectToPropertiesString Does indeed support arrays, this is exactly what https://github.com/NibiruChain/ts-sdk/blob/develop/src/gql/utils/consts.ts#L110 does

convertObjectToPropertiesString(
    {
        example: [
            {type: {deeper: "field"}}
        ],
        example2: {
            other: "field",
            this: "field"
        }
    }
)

Get's plainly converted to, which is proper GQL

example {
  type {
    deeper
  }
}
example2 {
  other
  this
}

Reply:
There really should only be one element per array when defining fields

If we want to limit that functionality in code we could do something like the following instead


```
    if (Array.isArray(value)) {
      const innerString = `${key} {
        ${convertObjectToPropertiesString(value[0] as IterableDictionary<T>)}
      }`
      result += `${innerString}\n`
    } else if (typeof value === "object" && value !== null) {
```

Reply:
After digging into this deeper ignore me, I see